elements/pfe-textinput/src/pfe-textinput.scss

```diff
@@ -0,0 +1,56 @@
+@import "../../pfe-sass/pfe-sass";
+
+$LOCAL: textinput;
+
+$LOCAL-VARIABLES: (
+  BackgroundColor: pfe-var(surface--lightest),
+  BackgroundColor--disabled: pfe-var(surface--base),
+  BorderColor: pfe-var(surface--border--lighter),
+  BorderBottomColor: pfe-var(surface--border--darker), // TODO: sync on this color #767676,
+  BorderBottomColor--hover: pfe-var(ui-accent),
+  BorderRadius: pfe-var(ui--border-radius),
+  Color: pfe-var(text),
+  FontSize: pfe-var(font-size),
+  Height: 36px,
+  Padding: 8px 6px,
+);
+
+:host {
+  display: inline-block;
+  position: relative;
+}
+
+::slotted(input) {
+  all: unset;
+  opacity: 0;
+  padding: pfe-local(Padding);
+  border-radius: pfe-local(BorderRadius);
+  border: pfe-local(BorderWidth) solid transparent;
+  height: pfe-local(Height) !important;
+}
+
+span input {
+  position: absolute;
+  z-index: 1;
+  top: 0;
+  left: 0;
+  box-sizing: border-box;
+  width: 100%;
+  padding: pfe-local(Padding);
+  border-radius: pfe-local(BorderRadius, $fallback: 0);
+  border: pfe-var(ui--border-width) pfe-var(surface--border-style)  pfe-local(BorderColor);
+  border-bottom-color: pfe-local(BorderBottomColor);
+  background-color: pfe-local(BackgroundColor);
+  font-size: pfe-local(FontSize);
+  color: pfe-local(color);
+
+  &:hover {
+    border-bottom-color: pfe-local(BorderBottomColor--hover);
+  }
+
+  &[disabled] {
```

Comment:
Should there also be styles for invalid field input?  Some kind of error state?

Reply:
There should be but I think we're opening up a much larger UX discussion. We could use a selector like `input:invalid`, but then we run into issues where, on page load, the input field would be considered invalid until the input field's conditions are met. For example, in this screenshot, the second field is required and shows as an error on page load because the field's conditions have not been met.

![image](https://user-images.githubusercontent.com/330256/91451944-fd6a1300-e84b-11ea-9430-fe9e0ac20fff.png)

Is that the user experience we want? I don't know. And yes, we could get a bit more sophisticated where we use JS to `checkValidity` on blur of a field and update then, but what about listening for a form submission. I think we may need to consider a `pfe-form` component that manages the state of it's children form element. But like I said, this is a much larger UX discussion.

Maybe for the time being, we expose a class, like `invalid` that could be added to the component wrapper that would visually indicate that something's wrong.

@castastrophe - What do you think?